deployment/puppet/cluster/files/q-agent-cleanup.py

```diff
@@ -477,6 +488,46 @@ def _reschedule_agent_l3(self, agent_type):
         self.log.info("_reschedule_agent_l3: ended rescheduling of orphaned routers")
         self.log.debug("_reschedule_agent_l3: end.")
 
+    def _reschedule_agent_lbaas(self, agent_type):
+        self.log.debug("_reschedule_agent_lbaas: start.")
+        alive_agents = []
+
+        # collect pool from removed lbaas agent
+        remove_pools = []
+        dead_agent = None
+        agent_host = socket.gethostname()
+        for agent in self._get_agents_by_type(agent_type):
+            if agent['alive'] and (agent_host != agent['host']):
+                self.log.info("found alive Lbaas agent: {0}".format(agent['id']))
+                alive_agents.append(agent)
+
+            if agent_host == agent['host']:
+                # dead agent
+                self.log.info("found removed Lbaas agent: {0}".format(agent['id']))
+                map(
+                    lambda pool: remove_pools.append(pool),
+                    self._list_pools_on_lbaas_agent(agent['id'])
+                )
+                dead_agent = agent
+
+        self.log.debug("Lbaas agents in cluster: {0}".format(alive_agents))
+        self.log.debug("Pools, attached to alive Lbaas agents: {0}".format(remove_pools))
+
+        if remove_pools and alive_agents:
+            add_agent = alive_agents[0]
```

Comment:
没有考虑随机是减少了判断是否存在不同的pool绑定相同vip的情况

2016-07-19 16:11 GMT+08:00 huntxu <notifications@github.com>:

> In deployment/puppet/cluster/files/q-agent-cleanup.py
> <https://github.com/eayunstack/fuel-library/pull/47#discussion_r71289598>:
>
> > +                alive_agents.append(agent)
> > +
> > +            if agent_host == agent['host']:
> > +                # dead agent
> > +                self.log.info("found removed Lbaas agent: {0}".format(agent['id']))
> > +                map(
> > +                    lambda pool: remove_pools.append(pool),
> > +                    self._list_pools_on_lbaas_agent(agent['id'])
> > +                )
> > +                dead_agent = agent
> > +
> > +        self.log.debug("Lbaas agents in cluster: {0}".format(alive_agents))
> > +        self.log.debug("Pools, attached to alive Lbaas agents: {0}".format(remove_pools))
> > +
> > +        if remove_pools and alive_agents:
> > +            add_agent = alive_agents[0]
>
> 是否考虑分流或者(伪)随机选择，而不是往一台agent上放所有失效的lb pool
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eayunstack/fuel-library/pull/47/files/34227f5f94dbd51594b544910280d08652a6d3d8#r71289598>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGUT0SUhbpIlEzjzI_eKcqjjAbu65Gvtks5qXIawgaJpZM4JOgK3>
> .
>